libbpfgo.go

```diff
@@ -297,33 +297,68 @@ func errptrError(ptr unsafe.Pointer, format string, args ...interface{}) error {
 }
 
 func NewModuleFromFile(bpfObjFile string) (*Module, error) {
+	return NewModuleFromFileBtf("", bpfObjFile)
+}
+
+func NewModuleFromFileBtf(btfObjPath string, bpfObjFile string) (*Module, error) {
 	C.set_print_fn()
-	bumpMemlockRlimit()
-	cs := C.CString(bpfObjFile)
-	obj := C.bpf_object__open(cs)
-	C.free(unsafe.Pointer(cs))
+	if err := bumpMemlockRlimit(); err != nil {
+		return nil, err
+	}
+	if btfObjPath == "" {
+		btfObjPath = "/sys/kernel/btf/vmlinux"
+	}
+	btfFile := C.CString(btfObjPath)
+	bpfFile := C.CString(bpfObjFile)
+
+	opts := C.struct_bpf_object_open_opts{}
+	opts.sz = C.sizeof_struct_bpf_object_open_opts
+	opts.btf_custom_path = btfFile
+
+	obj := C.bpf_object__open_file(bpfFile, &opts)
 	if C.IS_ERR_OR_NULL(unsafe.Pointer(obj)) {
 		return nil, errptrError(unsafe.Pointer(obj), "failed to open BPF object %s", bpfObjFile)
 	}
 
+	C.free(unsafe.Pointer(bpfFile))
+	C.free(unsafe.Pointer(btfFile))
+
 	return &Module{
 		obj: obj,
 	}, nil
 }
 
-func NewModuleFromBuffer(bpfObjBuff []byte, bpfObjName string) (*Module, error) {
+func NewModuleFromBuffer(btfObjPath string, bpfObjBuff []byte, bpfObjName string) (*Module, error) {
```

Comment:
Don't think we need to change the function signature here if not using `btfObjPath`